setup.py

```diff
@@ -1,34 +1,46 @@
-# -*- coding: utf-8 -*-
-"""
-    Setup file for kobuddy.
-    Use setup.cfg to configure your project.
+# see https://github.com/karlicoss/pymplate for up-to-date reference
 
-    This file was generated with PyScaffold 3.2.
-    PyScaffold helps you to put up the scaffold of your new Python project.
-    Learn more under: https://pyscaffold.org/
-"""
-import sys
 
-from pkg_resources import require, VersionConflict
-from setuptools import setup
+from setuptools import setup, find_packages # type: ignore
 
-try:
-    require('setuptools>=38.3')
-except VersionConflict:
-    print("Error: version of setuptools is too old (<38.3)!")
-    sys.exit(1)
 
-
-if __name__ == "__main__":
+def main():
+    pkgs = find_packages('src')
+    [pkg] = pkgs
     setup(
-        use_pyscaffold=True,
+        name=pkg,
+        use_scm_version={
+            'version_scheme': 'python-simplified-semver',
+            'local_scheme': 'dirty-tag',
+        },
+        setup_requires=['setuptools_scm'],
+
+        zip_safe=False,
+
+        packages=[pkg],
+        package_dir={'': 'src'},
+        package_data={pkg: ['py.typed']},
+
+        ## ^^^ this should be mostly automatic and not requiring any changes
```

Comment:
You could move all the information in setup.cfg and have a two-line setup.py
https://setuptools.readthedocs.io/en/latest/setuptools.html#configuring-setup-using-setup-cfg-files

Reply:
Thanks, but I find using `setup.cfg` pretty annoying because it's one extra file to maintain, and every time I need something slightly non-standard, I can't do it with setup.cfg and need to resort to setup.py anyway. For example when reusing dependencies/automatic documentation like [here](https://github.com/karlicoss/promnesia/blob/master/setup.py#L29-L38)

Reply:
I think that use case would be better covered by using `extras`!